### PR TITLE
Improvment: Improve thread poll

### PR DESCRIPTION
**Pull request by MaShizhen**

### CODE REVIEW

File: src/server/lib/SocketManager/index.ts [hunk 1]
```diff
@@ -13,18 +14,19 @@ const log = logger('service:socket-server');
...
+    sockets: Socket[] = [];
 
     events = [];
 
     start(server) {
         this.stop();
 
         this.server = server;
-        this.io = SocketIO(this.server, {
+        this.io = new Server(this.server, {
```
Comment (jane-rose) [inline src/server/lib/SocketManager/index.ts:27]: Leave a comment

**Review by jane-rose — APPROVED** [review]

File: src/server/lib/MeshProcess/MeshProcess.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ import {
     BOTTOM, FRONT, LEFT, RIGHT, TOP
 } from '../../constants';
 import { pathWithRandomSuffix } from '../../../shared/lib/random-utils';
-import DataStorage from '../../DataStorage';
+import global from '../global';
```
Comment (xiekun1992) [inline src/server/lib/MeshProcess/MeshProcess.js:10]: it is better not use `global` as variable name, just like `window` in browser, `global` is the root context for `node.js`, see: [Global objects](https://nodejs.org/dist/latest-v16.x/docs/api/globals.html#global)

File: src/server/lib/MeshProcess/STLToSvgStack.js [hunk 3]
```diff
@@ -6,7 +6,7 @@ import earcut from 'earcut';
 import * as THREE from 'three';
 import { MeshProcess } from './MeshProcess';
 import { pathWithRandomSuffix } from '../../../shared/lib/random-utils';
-import DataStorage from '../../DataStorage';
+import global from '../global';
```
Comment (xiekun1992) [inline src/server/lib/MeshProcess/STLToSvgStack.js:9]: it is better not use `global` as variable name, just like `window` in browser, `global` is the root context for `node.js`, see: [Global objects](https://nodejs.org/dist/latest-v16.x/docs/api/globals.html#global)

File: src/server/lib/editor/image-process.js [hunk 4]
```diff
@@ -2,7 +2,7 @@ import Jimp from '../jimp';
 
 import { pathWithRandomSuffix } from '../random-utils';
 import { convertRasterToSvg } from '../svg-convert';
-import DataStorage from '../../DataStorage';
+import global from '../global';
```
Comment (xiekun1992) [inline src/server/lib/editor/image-process.js:5]: it is better not use `global` as variable name, just like `window` in browser, `global` is the root context for `node.js`, see: [Global objects](https://nodejs.org/dist/latest-v16.x/docs/api/globals.html#global)

**Review by xiekun1992 — CHANGES_REQUESTED** [review]

File: src/server/lib/editor/image-process.js (hunk 4 above)
Comment (jane-rose) [reply src/server/lib/editor/image-process.js:5]: merge it now, change it later

**Comment by jane-rose** [comment]
https://snapmaker2.atlassian.net/browse/LU-2135

File: src/server/lib/MeshProcess/MeshProcess.js (hunk 2 above)
Comment (MaShizhen) [reply src/server/lib/MeshProcess/MeshProcess.js:10]: https://github.com/Snapmaker/Luban/pull/1460/files

File: src/server/lib/editor/image-process.js (hunk 4 above)
Comment (MaShizhen) [reply src/server/lib/editor/image-process.js:5]: https://github.com/Snapmaker/Luban/pull/1460/files

**Pull request merged**